src/main/java/com/infomaniak/lib/core/auth/CredentialManager.kt

```diff
@@ -17,15 +17,95 @@
  */
 package com.infomaniak.lib.core.auth
 
+import androidx.collection.ArrayMap
 import androidx.lifecycle.LiveData
+import com.facebook.stetho.okhttp3.StethoInterceptor
+import com.infomaniak.lib.core.BuildConfig
 import com.infomaniak.lib.core.models.user.User
+import com.infomaniak.lib.core.room.UserDatabase
+import com.infomaniak.lib.login.ApiToken
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import okhttp3.OkHttpClient
+import java.util.concurrent.TimeUnit
 
 /**
  * CredentialManager interface : Implement the essential methods to get Users and their Credentials to pass
  */
-interface CredentialManager {
+abstract class CredentialManager {
+
+    abstract var userDatabase: UserDatabase
+    abstract var currentUserId: Int
+    abstract var currentUser: User?
+
     /**
      * Get users, and their informations / tokens in a JSON format
      */
-    fun getAllUsers(): LiveData<List<User>>
-}
\ No newline at end of file
+    abstract fun getAllUsers(): LiveData<List<User>>
+
+    suspend fun setUserToken(user: User?, apiToken: ApiToken) {
+        user?.let {
+            it.apiToken = apiToken
+            userDatabase.userDao().update(it)
+        }
+    }
+
+    suspend fun getUserById(id: Int): User? = userDatabase.userDao().findById(id)
+
+
+    var onRefreshTokenError: ((user: User) -> Unit)? = null
```

Comment:
I think we could make an exception for this property, because it is important and I find that it is when a little hidden.
And so we could put it on top, so that it's quite visible.